### PR TITLE
Various new lints and type fixes

### PR DESCRIPTION
**Pull request by bhollis**

### CODE REVIEW

File: src/app/inventory/d2-stores.ts [hunk 1]
```diff
@@ -359,10 +359,7 @@ function processCharacter(
   const store = makeCharacter(defs, character, lastPlayedDate);
 
   // We work around the weird account-wide buckets by assigning them to the current character
-  const items = characterInventory.slice();
-  for (const k in characterEquipment) {
-    items.push(characterEquipment[k]);
-  }
+  const items = characterInventory.concat(characterEquipment.flat());
```
Comment (robojumper) [inline src/app/inventory/d2-stores.ts:362]: Does this `flat()` do anything?

Comment (bhollis) [reply src/app/inventory/d2-stores.ts:362]: It should? Each character inventory is a list of items.

File: src/app/bungie-api/rate-limit-config.ts [hunk 2]
```diff
@@ -1,36 +1,34 @@
-import { RateLimiterConfig, RateLimiterQueue } from './rate-limiter';
+import { addLimiter, RateLimiterQueue } from './rate-limiter';
 
 export default function setupRateLimiter() {
   // Bungie's API will start throttling an API if it's called more than once per second. It does this
```
Comment (nev-r) [inline src/app/bungie-api/rate-limit-config.ts:4]: unrelated, but.
start throttling a client?
start throttling an endpoint?

File: src/app/inventory/d2-stores.ts (hunk 1 above)
Comment (nev-r) [reply src/app/inventory/d2-stores.ts:362]: i'm with robo on this, either the types are wrong or this flat does nothing

**Review by nev-r — APPROVED** [review]

**Pull request merged**